tests/rdma/run.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/python3
+"""
+==========================================================================
+run.py - script for test client for Valkey Over RDMA (Linux only)
+--------------------------------------------------------------------------
+Copyright (C) 2024  zhenwei pi <pizhenwei@bytedance.com>
+
+This work is licensed under BSD 3-Clause, License 1 of the COPYING file in
+the top-level directory.
+==========================================================================
+"""
+import os
+import subprocess
+import netifaces
+import time
+
+def build_program():
+    cmd = "gcc tests/rdma/rdma-test.c -o tests/rdma/tmp/valkey-rdma-test -lrdmacm -libverbs -lpthread -g"
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    if p.wait():
+        print("Valkey Over RDMA building [FAILED]")
+        os._exit(1);
+
+    print("Valkey Over RDMA building [OK]")
+
+
+# iterate /sys/class/infiniband, find any usable RDMA device, and return IPv4 address
+def find_rdma_dev():
+    # Ex, /sys/class/infiniband/mlx5_0
+    # Ex, /sys/class/infiniband/rxe_eth0
+    # Ex, /sys/class/infiniband/siw_eth0
+    ibclass = "/sys/class/infiniband/"
+    for dev in os.listdir(ibclass):
+        # Ex, /sys/class/infiniband/rxe_eth0/ports/1/gid_attrs/ndevs/0
+        netdev = ibclass + dev + "/ports/1/gid_attrs/ndevs/0"
+        with open(netdev) as fp:
+            addrs = netifaces.ifaddresses(fp.readline().strip("\n"))
+            if netifaces.AF_INET not in addrs:
+                continue
+            ipaddr = addrs[netifaces.AF_INET][0]["addr"]
+            print("Valkey Over RDMA test prepare " + dev + " <" + ipaddr  + "> [OK]")
+            return ipaddr
+
+    return None
+
+
+# find any IPv4 available networking interface, setup soft RDMA of @provider
+def setup_soft_rdma(provider):
+    interfaces = netifaces.interfaces()
+    for interface in interfaces:
+        if interface == "lo":
+            continue
+
+        addrs = netifaces.ifaddresses(interface)
+        if netifaces.AF_INET not in addrs:
+            continue
+
+        softrdma = provider + "_" + interface
+        cmd = "rdma link add " + softrdma + " type " + provider + " netdev " + interface
+        p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        if p.wait():
+            print("Valkey Over RDMA setup " + softrdma + " [FAILED]")
+            continue
+
+        print("Valkey Over RDMA setup " + softrdma + " [OK]")
+        return softrdma
+
+    return None
+
+
+def test_rdma(ipaddr):
+    cwd = os.getcwd()
```

Comment:
If the script is started from another cwd, such as `cd tests/rdma; ./run.py`, this script doesn't work, right?

We can use something like this?

```python
    projectdir = os.path.dirname(os.path.abspath(__file__)) + "/../.."
```

Reply:
Pushed a [fix](https://github.com/valkey-io/valkey/pull/477/commits/b7f8cff3822bb626ae4490cf282b1f3f68314732), thanks a lot!